File: components/credentials/src/main/java/org/cloudfoundry/credhub/audit/entities/RegenerateCertificate.java
package org.cloudfoundry.credhub.audit.entities;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class RegenerateCertificate implements RequestDetails {
  private Boolean transitional;

  public Boolean getTransitional() {
    return transitional;
  }

  public void setTransitional(final Boolean transitional) {
    this.transitional = transitional;
  }

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.REGENERATE_CERTIFICATE;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/RegenerateCertificate.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

class RegenerateCertificate : RequestDetails {
    var transitional: Boolean? = null

    override fun operation(): OperationDeviceAction = OperationDeviceAction.REGENERATE_CERTIFICATE
}
